scipy/sparse/linalg/_special_sparse_arrays.py

```diff
@@ -511,3 +513,391 @@ def _adjoint(self):
 
     def _transpose(self):
         return self
+
+
+class Sakurai(LinearOperator):
+    """
+    Construct a Sakurai matrix in various formats and its eigenvalues.
+
+    Constructs the "Sakurai" matrix motivated by reference [1]_.
+    The matrix is square real symmetric positive definite and banded
+    with integer entries and analytically known eigenvalues.
+    The matrix gets ill-conditioned with its size growing.
+    See the notes below for details.
+
+    Parameters
+    ----------
+    n : int
+        The size of the matrix.
+    dtype : dtype
+        Numerical type of the array. Default is ``np.int32``.
+
+    Attributes
+    ----------
+    eigenvalues : ndarray, float
+        Eigenvalues of the Sakurai matrix  ordered ascending.
+
+    Methods
+    -------
+    toarray()
+        Construct a dense array from Laplacian data
+    tosparse()
+        Construct a sparse array from Laplacian data
+    tobanded()
+        The Sakurai matrix in the format for banded symmetric matrices,
+        i.e., (3, n) ndarray with 3 upper diagonals
+        placing the main diagonal at the bottom.
```

Comment:
This format matches what we have elsewhere in scipy for banded matrices, e.g.
https://github.com/scipy/scipy/blob/v1.11.2/scipy/linalg/_decomp_cholesky.py#L222-L233
but also https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.eig_banded.html etc. 👍

Reply:
Of course - this is where I have learned it from. What is your specific request here - adding these links?

Reply:
It was more to document for myself and the review that this is the standard format, because it wasn't clear (to me) from the formulation in the docstring. No changes needed.